wire-ios-sync-engine/Source/Use cases/AppendImageMessageUseCase.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAnalytics
+import WireDataModel
+
+// sourcery: AutoMockable
+public protocol AppendImageMessageUseCaseProtocol {
+
+    func invoke(
+        withImageData imageData: Data,
+        in conversation: ZMConversation
+    ) throws
+}
+
+public struct AppendImageMessageUseCase: AppendImageMessageUseCaseProtocol {
+
+    let analyticsSession: AnalyticsSessionProtocol?
+
+    public func invoke(
+        withImageData imageData: Data,
+        in conversation: ZMConversation
+    ) throws {
+        try conversation.appendImage(from: imageData)
+        analyticsSession?.trackEvent(
+            ConversationContributionAnalyticsEvent(
+                contributionType: .imageMessage,
+                conversationType: .init(
+                    conversation.conversationType
+                ),
+                conversationSize: UInt(
```

Comment:
why do we have to convert to UInt?

Reply:
“Our conversationSize property is defined as UInt because it represents a value that cannot be negative. Using UInt ensures that the conversationSize is always a non-negative number.”